src/core/tests/type_prop/convert_align_types.cpp

```diff
@@ -0,0 +1,297 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/op/convert_align_types.hpp"
+
+#include <gtest/gtest.h>
+
+#include "common_test_utils/test_assertions.hpp"
+#include "common_test_utils/type_prop.hpp"
+#include "openvino/op/parameter.hpp"
+
+using namespace ov;
+using namespace testing;
+
+struct ConvertAlignTypesTestParams {
+    PartialShape lhs_shape;
```

Comment:
I think shape can be removed as for parameter test the most important cases are all types and some bool flags which modify conversion rules.

check different shape in standard simple tests.

Reply:
As discussed offline, splited tests to multiple suites